website/addons/base/testing/views.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8 -*-
+import httplib as http
+import urlparse
+
+import mock
+from nose.tools import *  # noqa (PEP8 asserts)
+
+from framework.auth import Auth
+
+from website.addons.base.testing.base import OAuthAddonTestCaseMixin
+from website.util import api_url_for, web_url_for, permissions
+
+from tests.factories import AuthUserFactory, ProjectFactory
+
+
+class OAuthAddonAuthViewsTestCaseMixin(OAuthAddonTestCaseMixin):
+
```

Comment:
I don't think `test_dropbox_oauth_delete_user_with_invalid_credentials` is relevant anymore. We don't talk to dropbox (or any service) when deauthorizing the `ExternalAccount`, that's handled by `website/oauth/views.py:oauth_disconnect`, and it's all internal. It wouldn't fail if the creds were invalid because we don't need to know if the creds are valid or not to do the deauth.

As for `test_dropbox_oauth_finish_cancelled`, the test that handles this case is `test_user_denies_access` in `/tests/test_oauth.py`